app/views/job_invocations/_tab_overview.html.erb

```diff
@@ -47,7 +47,7 @@
       <b><%= _("Concurrency level limited to") %></b>: <%= job_invocation.concurrency_level %> <%= _('tasks at a time') %><br>
     <% end %>
     <% if job_invocation.time_span %>
-      <b><%= _("Set to distribute over") %></b>: <%= job_invocation.concurrency_level %> <%= _('minutes') %><br>
+      <b><%= _("Set to distribute over") %></b>: <%= job_invocation.time_span %> <%= _('minutes') %><br>
```

Comment:
We talk about seconds in the invocation form and minutes here, which one is correct?